src/Spryker/Client/RabbitMq/Model/Connection/ConnectionManager/ConnectionConfigFilter/ConnectionConfigFilter.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace Spryker\Client\RabbitMq\Model\Connection\ConnectionManager\ConnectionConfigFilter;
+
+use Spryker\Client\RabbitMq\Dependency\Client\RabbitMqToStoreClientInterface;
+use Spryker\Client\RabbitMq\Model\Connection\ConnectionFactoryInterface;
+
+class ConnectionConfigFilter implements ConnectionConfigFilterInterface
+{
+    /**
+     * @var \Spryker\Client\RabbitMq\Dependency\Client\RabbitMqToStoreClientInterface
+     */
+    protected $storeClient;
+
+    /**
+     * @var \Spryker\Client\RabbitMq\Model\Connection\ConnectionFactoryInterface
+     */
+    protected $connectionFactory;
+
+    /**
+     * @var bool[]|null
+     */
+    protected $connectionNameLocaleMap;
+
+    /**
+     * @param \Spryker\Client\RabbitMq\Dependency\Client\RabbitMqToStoreClientInterface $storeClient
+     * @param \Spryker\Client\RabbitMq\Model\Connection\ConnectionFactoryInterface $connectionFactory
+     */
+    public function __construct(RabbitMqToStoreClientInterface $storeClient, ConnectionFactoryInterface $connectionFactory)
+    {
+        $this->storeClient = $storeClient;
+        $this->connectionFactory = $connectionFactory;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QueueConnectionTransfer[] $connectionsConfig
+     * @param string|null $locale
+     *
+     * @return \Generated\Shared\Transfer\QueueConnectionTransfer[]
+     */
+    public function filterByLocale(array $connectionsConfig, ?string $locale): array
+    {
+        if ($locale === null) {
+            return $connectionsConfig;
+        }
+
+        if ($this->connectionNameLocaleMap === null) {
```

Comment:
Why not add this method to `ConnectionConfigFilter` and use it in this class instead?

Reply:
> It's not a static attribute, it will always be null, itdoesn't really make much sense to have the method addConnectionNameLocaleMap too, the method name is confusing anyway.
This method could be called some times in a row, and we need to check for null here just to prevent its calculation for each call. 

> Maybe you can even skip looping over the map array, and just
I am not sure that understand you here. `connectionNameLocaleMap` contains boolean values which say us about the existence of connection configuration for locales in all stores.

> Why not add this method to ConnectionConfigFilter and use it in this class instead?
Seems like your thoughts have flown far away and I lost the connection between your here.

I'll think about making this part cleaner. Thanks.

Reply:
I think I used the wrong class name in my comment, I meant another class but I forgot what it was :) anyway, It's much better and I think you got my point.